external/fv3fit/fv3fit/_shared/models.py

```diff
@@ -437,3 +437,64 @@ def load(cls, path: str) -> "OutOfSampleModel":
 
         model = cls(base_model, novelty_detector, cutoff=cutoff, taper=taper)
         return model
+
+
+@io.register("squashed_output_model")
+class SquashedOutputModel(Predictor):
+
+    _CONFIG_FILENAME = "squashed_output_model.yaml"
+
+    def __init__(
+        self,
+        base_model: Predictor,
+        squash_by_name: Optional[Hashable],
+        squash_threshold: Optional[float],
+        squash_to: Optional[float],
+    ):
+        if squash_by_name is not None:
+            if squash_threshold is None or squash_to is None:
+                raise ValueError(
+                    "If squash_by_name is specified, squash_threshold and squash_to "
+                    "must be also."
+                )
+            if squash_by_name not in base_model.output_variables:
+                raise ValueError(
+                    f"The squash by variable ({squash_by_name}) must among the "
+                    "set of model output variable names."
+                )
+
+        self._base_model = base_model
+        self._squash_by_name = squash_by_name
+        self._squash_threshold = squash_threshold
+        self._squash_to = squash_to
+
+        super().__init__(
+            input_variables=base_model.input_variables,
+            output_variables=base_model.output_variables,
+        )
+
+    def predict(self, X: xr.Dataset) -> xr.Dataset:
+        raw_predictions = self._base_model.predict(X)
+        if self._squash_by_name is not None:
+            squashed_predictions = raw_predictions.where(
```

Comment:
If there could be use cases where you don't want to squash everything according to the single `squash_by_name` output, I'd suggest using a pattern similar to what we've done in other models where they're initialized by a sequence of the desired transform / taper / etc configs (https://github.com/ai2cm/fv3net/blob/2a99e9db45966694db9d412dcb0887040fea0709/external/fv3fit/fv3fit/_shared/models.py#L285)

Reply:
If not, then could you make it clear in the docs that *everything* will be squashed according to the values of `squash_by_name`? Before getting to this part of the code, I thought that only that single field would be squashed.